src/abstractdataframe/abstractdataframe.jl

```diff
@@ -2173,3 +2173,209 @@ Base.getindex(::AbstractDataFrame, ::Union{Symbol, Integer, AbstractString}) =
 
 Base.setindex!(::AbstractDataFrame, ::Any, ::Union{Symbol, Integer, AbstractString}) =
     throw(ArgumentError("syntax df[column] is not supported use df[!, column] instead"))
+
+# insertcols!
+
+"""
+    insertcols!(df::AbstractDataFrame[, col], (name=>val)::Pair...;
+                makeunique::Bool=false, copycols::Bool=true)
+
+Insert a column into a data frame in place. Return the updated data frame.
+If `col` is omitted it is set to `ncol(df)+1`
+(the column is inserted as the last column).
+
+# Arguments
+- `df` : the data frame to which we want to add columns
+- `col` : a position at which we want to insert a column, passed as an integer
+  or a column name (a string or a `Symbol`); the column selected with `col`
+  and columns following it are shifted to the right in `df` after the operation
+- `name` : the name of the new column
+- `val` : an `AbstractVector` giving the contents of the new column or a value of any
+  type other than `AbstractArray` which will be repeated to fill a new vector;
+  As a particular rule a values stored in a `Ref` or a `0`-dimensional `AbstractArray`
+  are unwrapped and treated in the same way.
+- `makeunique` : Defines what to do if `name` already exists in `df`;
+  if it is `false` an error will be thrown; if it is `true` a new unique name will
+  be generated by adding a suffix
+- `copycols` : whether vectors passed as columns should be copied
+
+If `val` is an `AbstractRange` then the result of `collect(val)` is inserted.
+
+If `df` is a `SubDataFrame` then it must be created with `:` as column selector
+(otherwise an error is thrown). In this case the `copycols` keyword argument
+is ignored an added column is always copied and the parent data frame is
+filled with `missing` in rows that are filtered out by `df`.
```

Comment:
```suggestion
If `df` is a `SubDataFrame` then it must have been created with `:` as column selector
(otherwise an error is thrown). In this case the `copycols` keyword argument
is ignored (i.e. the added column is always copied) and the parent data frame's
column is filled with `missing` in rows that are filtered out by `df`.
```